### PR TITLE
Lint python Github Action.

### LINKED ISSUE #23: Fix  Lint python GitHub Action
Fix the following file: https://github.com/Karim-53/Compare-xAI/blob/main/.github/workflows/lint_python.yml
Current error: 
```
30    F821 undefined name 'itertools'
31
Error: Process completed with exit code 1.
```
Full error report: https://github.com/Karim-53/Compare-xAI/actions/runs/3860872221/jobs/6581483832

Documentation file that could help you fix the error: https://github.com/marketplace/actions/python-syntax-checker

### PR DESCRIPTION
**Pull request by Rishav1707**
## This PR closes the following issue:
closes #23 

## Changes made:
- Added the required run command in `lint-python.yml` file.

### CODE REVIEW

**Comment by Rishav1707** [comment]
@Karim-53 Please review my changes...

File: .github/workflows/lint_python.yml [hunk 1]
```diff
@@ -1,4 +1,3 @@
-# https://github.com/marketplace/actions/python-syntax-checker
 name: lint_python
```
Comment (Karim-53) [inline .github/workflows/lint_python.yml:1]:
```suggestion
# Source: https://github.com/marketplace/actions/python-syntax-checker
name: lint_python
```
Please keep the comment to help us retrieve the source of the Action

File: .github/workflows/lint_python.yml [hunk 2]
```diff
@@ -8,23 +7,15 @@ jobs:
       - uses: actions/checkout@v3
       - uses: actions/setup-python@v3
       - run: pip install bandit black codespell flake8 isort mypy pytest pyupgrade safety
-      # Security
-      # - run: bandit --recursive --skip B101 . || true  # B101 is assert statements
-
-      # Code style
-      # - run: black --check . || true
+      - run: bandit --recursive --skip B101 . || true  # B101 is assert statements
+      - run: black --check . || true
```
Comment (Karim-53) [inline .github/workflows/lint_python.yml:11]:
```suggestion
# Security
      # - run: bandit --recursive --skip B101 . || true  # B101 is assert statements

      # Code style
      # - run: black --check . || true
```
Let's keep this commented for now. We do not need this part of the Action

File: .github/workflows/lint_python.yml [hunk 3]
```diff
@@ -8,23 +7,15 @@ jobs:
...
-
-      # - run: safety check  # checks Python dependencies for known safety vulnerabilities and suggests the proper remediations for vulnerabilities detected.
-      
+      - run: isort --check-only --profile black . || true
+      - run: pip install -r requirements.txt || true
+      - run: mypy --install-types --non-interactive . || true
+      - run: pytest . || true
+      - run: pytest --doctest-modules . || true
+      - run: shopt -s globstar && pyupgrade --py36-plus **/*.py || true
+      - run: safety check
```
Comment (Karim-53) [inline .github/workflows/lint_python.yml:21]:
```suggestion
      # order libraries
      # - run: isort --check-only --profile black . || true

      # Unit tests
      # - run: pip install -r requirements.txt || true
      # - run: mypy --install-types --non-interactive . || true
      # - run: pytest . || true
      # - run: pytest --doctest-modules . || true
      # - run: shopt -s globstar && pyupgrade --py36-plus **/*.py || true

      # - run: safety check  # checks Python dependencies for known safety vulnerabilities and suggests the proper remediations for vulnerabilities detected.

```
Let's keep it commented for now

**Comment by Karim-53** [comment]
I double-checked, but I think the action is working correctly, actually. We should not change anything It is failing because of the .py script XD
Thanks for your time !

**Pull request closed without merging**